packages/manager/apps/dedicated/client/app/components/dedicated-cloud/users/vsphere-users/import/vsphere-user-import.controller.js

```diff
@@ -0,0 +1,108 @@
+import { DEFAULT_FILTER_COLUMN } from './vsphere-user-import.constant';
+
+export default class {
+  /* @ngInject */
+  constructor($q, $translate, DedicatedCloud) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.DedicatedCloud = DedicatedCloud;
+  }
+
+  $onInit() {
+    this.task = null;
+
+    this.loaders = {
+      init: true,
+      action: false,
+      poll: false,
+      cancelModal: false,
+    };
+
+    this.importUser = {
+      type: null,
+      userExemple: { user: null, group: null },
+      groupName: null,
+      userName: null,
+      domainName: null,
+      activeDirectoryId: null,
+    };
+
+    const params = {
+      offset: null,
+      pageSize: null,
+      sort: DEFAULT_FILTER_COLUMN,
+      sortOrder: 'ASC',
+      filters: [],
+      defaultFilterColumn: DEFAULT_FILTER_COLUMN,
+    };
+
+    return this.$q
+      .resolve(this.DedicatedCloud.getActiveDirectories(this.productId, params))
+      .then((federation) => {
+        this.federationList = federation;
+      })
+      .finally(() => {
+        this.loaders.init = false;
+      });
+  }
+
+  formIsValid() {
+    return (
+      this.importUser.type &&
+      (this.importUser.userName || this.importUser.groupName) &&
+      this.importUser.activeDirectoryId
+    );
+  }
+
+  onFederationChange(modelValue) {
+    this.importUser.userExemple.user = `${this.$translate.instant(
+      'dedicatedCloud_USER_import_user_exemple',
+    )}@${modelValue.domainName}`;
+    this.importUser.userExemple.group = `${this.$translate.instant(
+      'dedicatedCloud_USER_import_group_exemple',
+    )}@${modelValue.domainName}`;
+    this.importUser.activeDirectoryId = modelValue.activeDirectoryId;
+  }
+
+  onImportTypeChange(modelValue) {
+    this.importUser.type = modelValue;
+  }
+
+  postImportUser() {
+    this.loaders.action = true;
+
+    const params =
+      this.importUser.type === 'user'
+        ? { username: this.importUser.userName }
+        : { groupName: this.importUser.groupName };
+
+    return this.DedicatedCloud.postGrantActiveDirectory(
+      this.productId,
+      this.importUser.activeDirectoryId,
+      params,
+      this.importUser.type,
+    ).then(
+      (data) => {
+        this.task = data;
+      },
+      ({ data: err }) => {
+        this.loaders.action = false;
+        this.goBack(
+          `${this.$translate.instant('dedicatedCloud_USER_import_error')} ${
+            err.message
+          }`,
+          'danger',
+        );
+      },
+    );
+  }
+
+  closeModal() {
```

Comment:
nitpicking: 
I'm not too fond of refrerring this as `closeModal`, controller shouldn't be concerned about the way it displays the information (modal, list ...) but more about what data it will provide

Reply:
It's applied